disnake/emoji.py

```diff
@@ -247,15 +253,18 @@ async def delete(self, *, reason: Optional[str] = None) -> None:
         InvalidData
             The emoji data is invalid and cannot be processed.
         """
-        # this is an app emoji
-        if self.guild is None:
+        if self.guild_id is None:
+            # this is an app emoji
             if self.application_id is None:
                 # should never happen
-                msg = f"guild and application_id are both None when attempting to delete emoji with ID {self.id} This may be a library bug! Open an issue on GitHub."
+                msg = (
+                    f"guild_id and application_id are both None when attempting to delete emoji with ID {self.id}."
+                    " This may be a library bug! Open an issue on GitHub."
```

Comment:
This should be refactored into DiscordException for all errors raised by the library. Out of scope.